chartfx-chart/src/test/java/de/gsi/chart/ui/css/CssPropertyFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * @author akrimm
+ */
+@ExtendWith(ApplicationExtension.class)
+@ExtendWith(SelectiveJavaFxInterceptor.class)
+public class CssPropertyFactoryTest {
+    private final TestStyleable styleable = new TestStyleable();
+
+    @Start
+    public void start(Stage stage) {
+        // we need to add the styleable to a scene and display it or the css will not be evsaluated
+        final VBox root = new VBox(styleable);
+        final Scene scene = new Scene(root, 100, 120);
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    @Test
+    public void testApplyingStyle(FxRobot fxRobot) {
+        verifyThat(styleable.getMinHeight(), equalTo(-1.0)); // parent property
+        verifyThat(styleable.getTestDouble(), equalTo(4.2));
+        verifyThat(styleable.getTestInt(), equalTo(42));
+        verifyThat(styleable.getTestString(), equalTo("test"));
+        verifyThat(styleable.getTestBool(), equalTo(false));
+        verifyThat(styleable.getTestObject(), equalTo(Color.PINK));
+        verifyThat(styleable.getTestEnum(), equalTo(TestEnum.ENABLED));
+        verifyThat(styleable.getTestEnumP(), equalTo(TestEnum.DISABLED));
+        verifyThat(styleable.getPseudoClassStates(), Matchers.contains(PseudoClass.getPseudoClass("disabled")));
+        verifyThat(styleable.getPseudoClassStates(), not(contains(PseudoClass.getPseudoClass("enabled"))));
+        fxRobot.interact(() -> styleable.setStyle("-fx-test-double: 2.3; -fx-test-int: 23; -fx-test-string: styletest;-fx-test-bool: true;-fx-test-object: green;-fx-test-enum: disabled;-fx-test-enum-p: enabled; -fx-min-height: 13;"));
+        verifyThat(styleable.getMinHeight(), equalTo(13.0)); // parent property
+        verifyThat(styleable.getTestDouble(), equalTo(2.3));
+        verifyThat(styleable.getTestObject(), equalTo(Color.GREEN));
+        verifyThat(styleable.getTestInt(), equalTo(23));
+        verifyThat(styleable.getTestString(), equalTo("styletest"));
+        verifyThat(styleable.getTestEnum(), equalTo(TestEnum.DISABLED));
+        verifyThat(styleable.getTestEnumP(), equalTo(TestEnum.ENABLED));
+        verifyThat(styleable.getTestBool(), equalTo(true));
+        verifyThat(styleable.getPseudoClassStates(), not(contains(PseudoClass.getPseudoClass("disabled"))));
+        verifyThat(styleable.getPseudoClassStates(), contains(PseudoClass.getPseudoClass("enabled")));
+
+        // test applying via css sheet
+        // test with multiple instances
+    }
+
+    public static class TestStyleable extends Region {
+        private static final CssPropertyFactory<TestStyleable> CSS = new CssPropertyFactory<>(Region.getClassCssMetaData());
+
+        @Override
+        public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
+            return CSS.getCssMetaData();
+        }
+
+        // double
+        private final StyleableDoubleProperty testDouble = CSS.createDoubleProperty(this, "testDouble", "-fx-test-double", s -> s.testDouble, 4.2, true, null);
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5741094)